src/clean-core/to_string.cc

```diff
@@ -95,3 +141,834 @@ cc::string cc::to_string(std::byte value)
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%#hX", value);
+                ss << cc::string_view(buffer, length);
+            }
+            if constexpr (sizeof(IntType) == 4)
+            {
+                char buffer[10 + 1 + 1];
+                // the cast prevents a warning on MSVC where 32 bit integer are unsigned long
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%#X", static_cast<unsigned int>(value));
+                ss << cc::string_view(buffer, length);
+            }
+            if constexpr (sizeof(IntType) == 8)
+            {
+                char buffer[19 + 1 + 1];
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%#llX", value);
+                ss << cc::string_view(buffer, length);
+            }
+        }
+        else
+        {
+            if constexpr (sizeof(IntType) == 1)
+            {
+                char buffer[3 + 1 + 1];
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%hhX", value);
+                ss << cc::string_view(buffer, length);
+            }
+            if constexpr (sizeof(IntType) == 2)
+            {
+                char buffer[6 + 1 + 1];
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%hX", value);
+                ss << cc::string_view(buffer, length);
+            }
+            if constexpr (sizeof(IntType) == 4)
+            {
+                char buffer[10 + 1 + 1];
+                // the cast prevents a warning on MSVC where 32 bit integer are unsigned long
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%X", static_cast<unsigned int>(value));
+                ss << cc::string_view(buffer, length);
+            }
+            if constexpr (sizeof(IntType) == 8)
+            {
+                char buffer[19 + 1 + 1];
+                auto const length = std::snprintf(buffer, sizeof(buffer), "%llX", value);
+                ss << cc::string_view(buffer, length);
+            }
+        }
+    }
+    break;
+    default:
+        CC_ASSERT(false && "Invalid format string: Unsupported argument type for int type");
+    }
+}
+
+template <class IntType>
+void int_to_string_impl(cc::string_stream& ss, IntType value, parsed_fmt_args const& args)
+{
+    auto const is_neg = value < 0;
+    std::make_unsigned_t<IntType> unsigned_value = is_neg ? -value : value;
+
+    switch (args.sign)
+    {
+    case ' ':
+        ss << (is_neg ? "-" : " ");
+        break;
+    case '-':
+        if (is_neg)
+            ss << "-";
+        break;
+    case '+':
+        ss << (is_neg ? "-" : "+");
+        break;
+    }
+    unsigned_to_string_impl(ss, unsigned_value, args);
+}
+
+template <class FloatType>
+void to_string_float_impl(cc::string_stream& ss, FloatType value, parsed_fmt_args const& args)
+{
+    // note: currently completely relys on std::sprintf
+
+    cc::string_stream sprintf_args;
```

Comment:
Seems to me that char capped_vector or similar could do this job more efficiently

Reply:
it's not _that_ easy because later a `to_string(sprintf_args, args.width, "");` is used

Reply:
Also I plan to get if all calls to `std::sprintf`, to the implementation is bound to change quite significantly :)

Reply:
Yes, the API really would benefit from stream_ref here (although this particular case could just use another snprintf)